app/models/user_project_role.rb

```diff
@@ -0,0 +1,15 @@
+class UserProjectRole < ActiveRecord::Base
+  include HasProjectRole
+  belongs_to :project
+  belongs_to :user
+
+  validates_presence_of :project, :user
+  validates :role_id, inclusion: { in: ProjectRole.all.map(&:id) }
+  validates_uniqueness_of :project_id, scope: :user_id
+
+  def self.update_user_role(id, attributes)
```

Comment:
Ok. I'll move it back to the controller if you guys prefer to see this in the controller.